tasks/benchmark-install-against-revisions/src/work_env.rs

```diff
@@ -0,0 +1,175 @@
+use crate::fixtures::{CLEANUP_SCRIPT, INSTALL_SCRIPT, PACKAGE_JSON};
+use itertools::Itertools;
+use os_display::Quotable;
+use pipe_trait::Pipe;
+use std::{
+    fs::{self, File, Permissions},
+    io::Write,
+    iter,
+    path::{Path, PathBuf},
+    process::{Command, Stdio},
+};
+
+#[derive(Debug)]
+pub struct WorkEnv {
+    pub root: PathBuf,
+    pub revisions: Vec<String>,
+    pub registry: String,
+    pub repository: PathBuf,
+}
+
+impl WorkEnv {
+    fn root(&self) -> &'_ Path {
+        &self.root
+    }
+
+    fn revisions(&self) -> impl Iterator<Item = &'_ str> + '_ {
+        self.revisions.iter().map(AsRef::as_ref)
+    }
+
+    fn registry(&self) -> &'_ str {
+        &self.registry
+    }
+
+    fn repository(&self) -> &'_ Path {
+        &self.repository
+    }
+
+    fn revision_root(&self, revision: &str) -> PathBuf {
+        self.root().join(revision)
+    }
+
+    fn revision_cleanup_script(&self, revision: &str) -> PathBuf {
+        self.revision_root(revision).join("cleanup.bash")
+    }
+
+    fn revision_install_script(&self, revision: &str) -> PathBuf {
+        self.revision_root(revision).join("install.bash")
+    }
+
+    fn revision_repo(&self, revision: &str) -> PathBuf {
+        self.revision_root(revision).join("pacquet")
+    }
+
+    fn init(&self) {
+        const INIT_PROXY_CACHE: &str = ".init-proxy-cache";
+
+        eprintln!("Initializing...");
+        for revision in self.revisions().chain(iter::once(INIT_PROXY_CACHE)) {
+            eprintln!("Revision: {revision:?}");
+            let dir = self.revision_root(revision);
+            fs::create_dir_all(&dir).expect("create directory for the revision");
+            fs::write(dir.join("package.json"), PACKAGE_JSON)
+                .expect("create package.json for the revision");
+            create_script(&self.revision_cleanup_script(revision), CLEANUP_SCRIPT);
+            create_script(&self.revision_install_script(revision), INSTALL_SCRIPT);
+            create_npmrc(&dir, self.registry());
+        }
+
+        eprintln!("Populating proxy registry cache...");
+        Command::new("pnpm")
+            .current_dir(self.revision_root(INIT_PROXY_CACHE))
+            .arg("install")
+            .pipe(executor("pnpm install"));
+    }
+
+    fn build(&self) {
+        eprintln!("Building...");
+        for revision in self.revisions() {
+            eprintln!("Revision: {revision:?}");
+
+            let repository = self.repository();
+            let revision_repo = self.revision_repo(revision);
+            if revision_repo.exists() {
+                eprintln!("Updating {revision_repo:?} to upstream...");
+                Command::new("git")
+                    .current_dir(&revision_repo)
+                    .arg("pull")
+                    .arg("origin")
+                    .arg(revision)
+                    .pipe(executor("git pull"));
```

Comment:
This command sometimes spawn a commit editor. I want to prevent that. Do you guys have any idea?